### DISCUSSION TITLE
Graphics libraries and HardDisk based iCub - mount the icub-head filesystem on client

### DISCUSSION THREAD

**Discussion by mbrunettini** [robotology/community#184] [Announcements]
**On the new PC104 with hard disk** (a.k.a. _icub-head_), graphics-based tools have not been installed (except the ones required for firmware interfaces) and it is better not to install them. Anyway, to use graphic tools (like editors, files managers, etc..) is very useful and sometimes essential.
Instead of running remotely a tool and displaying it locally (like we did in old PC104), we can **mount locally via SSH the icub-head filesystem** and run locally all the tools we need.
In order to mount locally the icub-head filesystem _on your client_: 
1) install the _sshfs_ package 
`sudo apt-get install sshfs`
2) create a mount point for the remote filesystem 
(eg.) `mkdir /home/icub/icub-head_fs` 
3) mount the remote /usr/local/src/robot via SSH 
`sudo sshfs -o allow_other icub@10.0.0.2:/usr/local/src/robot/ /home/icub/icub-head_fs`

if you already setup the passwordless SSH login, you can skip the password:
`sudo sshfs -o allow_other,IdentityFile=/home/icub/.ssh/id_rsa  icub@10.0.0.2:/usr/local/src/robot/ /home/icub/icub-head_fs`

**Comment by mbrunettini** [comment]
I will add a command alias on the client machine icub bash environment, so the process will be simpler.

**Comment by traversaro** [comment]
cc @robotology/icub-developers

**Comment by S-Dafarra** [comment]
Thanks for the guide! I was wondering, what is the correct procedure to unmount the remote from the client once I'm done working on it?

**Comment by francesco-romano** [comment]
@mbrunettini what about adding a line in `/etc/fstab` for that mount point?
Something like:

```
sshfs#icub@10.0.0.2:/usr/local/src/robot/  /mnt/icub-head   fuse    defaults,allow_other    0  0
```
In this way, mounting and unmounting is performed through the `mount` utility.

Then, you are more expert than me regarding the proper options to put in fstab, like `noauto` or `user`, etc...(like, disabling automatic mount at boot, or allowing users without sudo)

**Comment by mbrunettini** [comment]
The unmount procedure is the same as a normal mounted filesystem:
`sudo umount _MOUNT_POINT_ `
or 
`sudo fusermount -u _MOUNT_POINT_ `
